llama-index-integrations/readers/llama-index-readers-wikipedia/llama_index/readers/wikipedia/base.py

```diff
@@ -27,15 +28,22 @@ def __init__(self) -> None:
     def class_name(cls) -> str:
         return "WikipediaReader"
 
-    def load_data(self, pages: List[str], **load_kwargs: Any) -> List[Document]:
+    def load_data(
+        self, pages: List[str], lang_code: str = "en", **load_kwargs: Any
+    ) -> List[Document]:
         """Load data from the input directory.
 
         Args:
             pages (List[str]): List of pages to read.
-
+            lang_code (str): Language code for Wikipedia. Defaults to English. Valid Wikipedia language codes
+            can be found at https://en.wikipedia.org/wiki/List_of_Wikipedias.
         """
         import wikipedia
 
+        if lang_code.lower() != "en":
+            # Sets, without checking the validity of, the language code for Wikipedia.
```

Comment:
can we add some validation here? It looks like we can (and maybe should) check and raise error if the supplied `lang_code` isn't actually supported via `wikipedia.languages()` 

[source](https://wikipedia.readthedocs.io/en/latest/code.html#wikipedia.languages)

Reply:
@nerdai thanks for your comment. That sounds like a good idea. How about the following?

```
if lang_code.lower() in wikipedia.languages().keys():
    wikipedia.set_lang(lang_code.lower())
else:
    raise SomeError(with_a_message)
```

What error would you like to raise? `wikipedia.exceptions.WikipediaException`? The error message should be something like "The provided language prefix for Wikipedia is not supported. Check supported languages at https://en.wikipedia.org/wiki/List_of_Wikipedias."

Alternatively, would you like it to silently fall back to "en" if the prefix is not supported instead of raising an exception?

Since I use the word prefix instead of code in the error message to be consistent with the `wikipedia` package, I will also change the function argument from `lang_code` to `lang_prefix`.

Looks good?

Reply:
Yea, we can raise a `ValueError` here. The message you suggested seems reasonable to me. 

`lang_prefix` > `lang_code` 🙏

Thanks!

Reply:
Thanks, I think the following is better than checking `wikipedia.languages().keys()`:

```
if lang_prefix.lower() in wikipedia.languages():
                wikipedia.set_lang(lang_prefix.lower())
            else:
                raise ValueError(
                    f"Language prefix '{lang_prefix}' for Wikipedia is not supported. Check supported languages at https://en.wikipedia.org/wiki/List_of_Wikipedias."
                )
```

Reply:
Agreed!

Reply:
@nerdai thanks, code pushed.